test/suites/host-ip-change.robot

```diff
@@ -0,0 +1,233 @@
+    [RETURN]    ${nodeName}
+
+Modify NetworkManager Configuration
+    [Arguments]    ${ip}    ${gateway}   ${device}
+    [Documentation]    Modify NetworkManager configuration using nmcli commands
+    ${stdout}   ${rc1}=    Execute Command    nmcli con modify ${device} ipv4.addresses ${ip}
+    ...    sudo=True    return_rc=True
+    ${stdout}   ${rc2}=    Execute Command    nmcli con mod ${device} ipv4.gateway ${gateway}
+    ...    sudo=True    return_rc=True
+    ${stdout}   ${rc3}=    Execute Command    nmcli con mod ${device} ipv4.method manual
+    ...    sudo=True    return_rc=True
+    Should Be Equal As Integers    ${rc1}     0
+    Should Be Equal As Integers    ${rc2}     0
+    Should Be Equal As Integers    ${rc3}     0
+
+Login MicroShift Host With New IP
+  [Arguments]  ${ip}
+  [Documentation]  Login to the MicroShift host via ssh and leave the connection open
+  ...  This keyword is meant to be used at the suite level. This ensures
+  ...  most tests already have an open connection. Any tests that will take
+  ...  action that disrupt that connection are responsible for restoring it.
+
+  Log  Host: ${ip}
+  IF    '${SSH_PORT}'
+        SSHLibrary.Open Connection    ${ip}    port=${SSH_PORT}
+  ELSE
+        SSHLibrary.Open Connection    ${ip}
+  END
+  Wait And Retry Login MicroShift Host  ${ip}
+  # If there is an ssh key set in the global configuration, use that to
+  # login to the host. Otherwise, assume that the ssh agent is running
+  # and configured properly.
+  IF  '${SSH_PRIV_KEY}'
+    SSHLibrary.Login With Public Key  ${USHIFT_USER}  ${SSH_PRIV_KEY}
+  ELSE
+    SSHLibrary.Login  ${USHIFT_USER}  allow_agent=True
+  END
+
+Wait And Retry Login MicroShift Host
+  [Arguments]  ${ip}
+  [Documentation]  Wait and retry logging in to the MicroShift host via SSH
+   ...  with a specified maximum number of retries and interval between retries.
+  Wait Until Keyword Succeeds  20x  15s  SSHLibrary.Login With Public Key  ${USHIFT_USER}  ${SSH_PRIV_KEY}
+
+Reboot Microshift System
+    [Documentation]    Reboot the system using "systemctl reboot" command
+    ${stdout}    ${rc}=    Execute Command    systemctl reboot
+    ...    sudo=True    return_rc=True
+    Log    Reboot command executed. Return code: ${rc}
+    Run Keyword IF    ${rc} != 0    Fail    Reboot command failed with return code ${rc}
+
+Verify Microshift Show Config Command Output
+    [Arguments]  ${new_host_ip}
+    [Documentation]   Verify that 'microshift show-config' command output contains the IP address '10.1.235.91'
+    ${stdout}    ${rc}=    Execute Command    microshift show-config
+    ...    sudo=True    return_rc=True
+    Log    'microshift show-config' command output: ${stdout}
+    Should Contain    ${stdout}    ${new_host_ip}
+    Should Be Equal As Integers    ${rc}    0
+
+Check New Kubeconfig Generated For New IP
+    [Documentation]    Check the new kubeconfig
+    ${hostname}   ${rc1}=    Execute Command    hostname
```

Comment:
This is looking for the file using the hostname, but the test does not change the hostname.

Reply:
@dhellmann When we change ip it generated new config in new hostname directory and also hostname get changed automatically

Reply:
I have not seen VMs change name when their IPs change. What causes that?

Reply:
I'm not sure about this.

Reply:
@dhellmann The only explanation I can come up with is we (QE at least) are using a transient hostname rather than a static hostname on the MicroShift cluster. The output of **hostnamectl** confirms this. The kickstart file that gets embedded in the ISO is using DHCP for the interface. 

[hostname info](https://www.computernetworkingnotes.com/linux-tutorials/how-to-change-the-hostname-on-linux.html)

Reply:
Is that expectation holding in CI?

I'd like the tests to be explicit about expectations, so let's add a check that the hostname value changes after the IP changes. If we can't ensure that in CI, then we need the test to be looking for the cert for the IP address or some other signal that MicroShift has done some work based on the IP changing.